src/st/find_entity_horizontal.h

```diff
@@ -11,13 +11,13 @@
  *    x position larger than `posX` or the tail eleement.
  */
 void FindFirstEntityToTheRight(s16 posX) {
-    while (true) {
-        if (g_LayoutObjHorizontal->posX != LAYOUT_OBJ_START &&
-            g_LayoutObjHorizontal->posX >= posX) {
+    while (1) {
+        u16* layoutEntity = g_LayoutObjHorizontal;
+        if (layoutEntity[0] != LAYOUT_OBJ_START && layoutEntity[0] >= posX) {
             break;
         }
 
-        g_LayoutObjHorizontal++;
+        g_LayoutObjHorizontal += sizeof(LayoutEntity) / sizeof(u16);
```

Comment:
Why keep LayoutEntity here when the rest of the PR removes the structs?

Reply:
It is to avoid using the magical number `+= 5` or `-= 5`. The array is still aligned by the struct `LayoutEntity`, which is still getting propagated. I also use it in 6FD0 to declare the data. I am still undecided if to remove `LayoutEntity` completely... I was forced to use a `u16*` for these functions just due to the PSP match.

Let's continue with the project and make any further adjustments accordingly!